### PR TITLE
Bulk dev_guide from devel to 2.3

### PR DESCRIPTION
**Pull request by gundalow**
##### SUMMARY
Bulk import of relevant `dev_guide` docs from `devel` to `2.3`

##### ISSUE TYPE
 - Docs Pull Request

### CODE REVIEW

File: docs/docsite/rst/dev_guide/developing_modules.rst [hunk 1]
```diff
@@ -31,17 +31,27 @@ develop a module. Ask the following questions:
 
 1. Does a similar module already exist?
 
-There are a lot of existing modules available, and more that are in development. You should check out the list of existing modules at :doc:`../modules` or look at the `module PRs <https://github.com/ansible/ansible/labels/module>`_ for the ansible repository on Github to see if a module that does what you want exists or is in development.
+There are a lot of existing modules available, you should check out the list of existing modules at :doc:`../modules`
```
Comment (dharmabumstead) [inline docs/docsite/rst/dev_guide/developing_modules.rst]: No comma, please - semicolon or new sentence.

File: docs/docsite/rst/dev_guide/developing_modules.rst [hunk 2]
```diff
@@ -31,17 +31,27 @@ develop a module. Ask the following questions:
...
 
-2. Should you use or develop an action plugin instead?
+2. Has someone already worked on a similar Pull Request?
```
Comment (dharmabumstead) [inline docs/docsite/rst/dev_guide/developing_modules.rst]: Don't capitalize "pull request".

File: docs/docsite/rst/dev_guide/developing_modules.rst [hunk 3]
```diff
@@ -31,17 +31,27 @@ develop a module. Ask the following questions:
...
+
+It's possible that someone has already started developing a similar PR. There are a few ways to find open module Pull Requests:
```
Comment (dharmabumstead) [inline docs/docsite/rst/dev_guide/developing_modules.rst]: Ibid.

**Review by dharmabumstead — DISMISSED** [review]
Still need to review the new testing content.

**Comment by gundalow** [comment]
Testing content was reviewed when it was originally written and put into `devel`, this PR is just copying the content into `stable-2.3`

**Comment by dharmabumstead** [comment]
Thanks @gundalow!

**Pull request merged**